internal/martian/log/log.go

```diff
@@ -30,18 +35,31 @@ func SetLogger(l Logger) {
 	currLogger = l
 }
 
+type contextKey string
+
+const TraceContextKey contextKey = "trace"
+
 // Infof logs an info message.
-func Infof(format string, args ...any) {
+func Infof(ctx context.Context, format string, args ...any) {
+	if v := ctx.Value(TraceContextKey); v != nil {
+		format = fmt.Sprintf("[%s] %s", v, format)
+	}
```

Comment:
This is repeated 3 times, maybe extract that to function?

Reply:
ok